### DISCUSSION TITLE
How to hide the cursor?

### DISCUSSION THREAD

**Discussion by amSnoop** [AvaloniaUI/Avalonia#11675] [Q&A]
I am making something that uses VLCSharp to render video, and I want to hide the cursor. I found [this](https://github.com/radiolondra/libvlcsharp/tree/3.x/src/LibVLCSharp.Avalonia.Unofficial) and am using it to add a window over the video player, populated with an empty, transparent canvas. I have two issues: First, using `Overlay.Cursor = new Cursor(StandardCursorType.None);` doesn't do anything. Setting it to something like `Hand` does, but not `None`. I tried to work around this by making a transparent cursor image, but that's my second issue: The transparent image shows up as a gray box instead of being transparent. The repo where this issue is is [here](https://github.com/amSnoop/DeLightNew/). Specifically, in `VideoWindow.axaml.cs` on line 58.

ETA: I am using Win11, .NET 7.0, LibVLCSharp 3.7, and Avalonia ~~10.21~~ 11 rc-1

Edit: Workaround: Use a translucent image for the cursor with an alpha value as low as possible. It will be perceptually invisible.

**Comment by timunie** [comment]
There was an issue with some cursors, but I think it was only fixed for 11.0 versions. Don't know if you can upgrade to 11.0 rc1 and try again?

**Reply by timunie** [reply]
Also I'm not sure if that particular issue was affected or not.

**Reply by amSnoop** [reply]
Ok, I updated, but no effect. Neither attempt at hiding the cursor seems to have changed. Still a gray dot instead of transparent image, and `None` still has no effect.

**Reply by timunie** [reply]
what nuget feed do I have to use to pull `LibVLCSharp.Avalonia.Unofficial` ? I cannot find this package on nuget

**Reply by amSnoop** [reply]
Apologies. I have removed the reference to it. It was used to get me the `VideoView.cs` file, but the more complicated nature of needing to download then install the package led me to just copy the code into my own project, since it was only one file. I just never removed the reference. Please try pulling again.

Thanks

**Reply by amSnoop** [reply]
It originally came from here:

https://github.com/radiolondra/libvlcsharp

**Comment by amSnoop** [comment]
I tried it again with a fresh Avalonia 11.0 rc-1 project, and `None` still doesnt work

```csharp
using Avalonia.Controls;
using Avalonia.Input;

namespace CursorHidingTest {
    public partial class MainWindow : Window {
        public MainWindow() {
            InitializeComponent();
            Cursor = new(StandardCursorType.Hand);
        }
    }
}
```
This is the only thing I changed when making a fresh project, and it works exactly as I expect it to, just not if you set it to `None`

Further testing with transparent images reveals that the "gray box" issue only presents itself if the entire image is transparent. If I add a single black pixel to an otherwise transparent image, then the transparency is properly rendered. 

Is there another workaround I can use to try and achieve this effect for the time being? These seem like they might be larger issues (or in the case of the transparent image, possibly intended behaviors)

For reference, the code that I used for the transparent image:


```csharp

using Avalonia;
using Avalonia.Controls;
using Avalonia.Input;
using Avalonia.Media.Imaging;

namespace CursorHidingTest {
    public partial class MainWindow : Window {
        public MainWindow() {
            InitializeComponent();
            var transCur = new Cursor(new Bitmap("C:\\Untitled.png"), new PixelPoint(0, 0));
            Cursor = transCur;
        }
    }
}

```

**Comment by amSnoop** [comment]
I found a workaround: Use a translucent image. I made a 1x1 pixel png that has an RGBA value of `00` `00` `00` `01` and used that as my cursor image. The image is rendered properly and the alpha is so low that it is perceptually indistinguishable from being invisible. See my previous comment for the code I used to set the cursor to that image. 

This still seems like some kind of bug with the `StandardCursorType.None` functionality, but at least for the time being, I can use this work-around.

**Comment by joelphilippage** [comment]
The workaround isn't working for me on the latest version, neither is setting cursor to none. I had to use the following code on windows:
`[DllImport("user32.dll")]
static extern int ShowCursor(bool bShow);

ShowCursor(false);`

**Comment by ExaurovNM** [comment]
If you want to hide cursor over video content (VLC)
then use VLC's parameter **--mouse-hide-timeout=0** 
`LibVLC("--mouse-hide-timeout=0")`
